jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -38,19 +38,18 @@
 import org.eclipse.jetty.http.HttpField;
 import org.eclipse.jetty.http.HttpHeader;
 import org.eclipse.jetty.http.QuotedCSV;
-import org.eclipse.jetty.util.StringUtil;
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
 
 public abstract class AuthenticationProtocolHandler implements ProtocolHandler
 {
     public static final int DEFAULT_MAX_CONTENT_LENGTH = 16*1024;
     public static final Logger LOG = Log.getLogger(AuthenticationProtocolHandler.class);
-    
-    private static final Pattern PARAM_PATTERN = Pattern.compile("([^=]+)=(.*)");
-    private static final Pattern TYPE_PATTERN = Pattern.compile("([^\\s]+)(\\s+(.*))?");
-    private static final Pattern MULTIPLE_CHALLENGE_PATTERN = Pattern.compile("(.*?)\\s*,\\s*([^=\\s,]+(\\s+[^=\\s].*)?)");
-    private static final Pattern BASE64_PATTERN = Pattern.compile("[\\+\\-\\.\\/\\dA-Z_a-z~]+=*");
+
+    private enum State { AUTH_SCHEME, TOKEN68, AUTH_PARAMS, CHALLENGE_END }
```

Comment:
I don't think we need all these states... actually we might not need any states.

The first item in the CSV list, must be one of
 + SCHEME
 + SCHEME ' ' PARAM
 + SCHEME ' ' TOKEN68

Then for all other entries in the CSV list they are one of:
 + PARAM
 + SCHEME
 + SCHEME ' ' PARAM
 + SCHEME ' ' TOKEN68

So I think the only state we really need to keep between items in the list is the last HeaderInfo created (to which we add more PARAMs)